gitclone/checkout_simple_strategy.go

```diff
@@ -0,0 +1,129 @@
+package gitclone
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+type checkoutStrategy interface {
+	Validate() error
+	Do(gitCmd git.Git) *step.Error
+}
+
+//
+// checkoutNone
+type checkoutNone struct {
+	ShouldUpdateSubmodules bool
+}
+
+func (c checkoutNone) Validate() error {
+	return nil
+}
+
+func (c checkoutNone) Do(gitCmd git.Git) *step.Error {
+	if c.ShouldUpdateSubmodules {
+		return updateSubmodules(gitCmd)
+	}
+
+	return nil
+}
+
+//
+// checkoutCommit
+type checkoutCommit struct {
```

Comment:
pls separate checkout strategies params. eg have commitCheckoutParams, and so on. so later in webhook handler, we can import these params and return them, rather than a generic param struct. This way we can make sure git-clone receives valid input combinations.